### ISSUE TITLE
Racket unsafe-fx functions missing in bootstrap/idris2_app/idris2.rkt

### ISSUE THREAD

**Issue by jfoutz** [idris-lang/Idris2#2279]
If possible, please include complete self contained source files (for example by uploading to https://gist.github.com) that exhibit the issue in addition to quoting from them here. The smaller the example the better!
docs have updated location - https://docs.racket-lang.org/reference/unsafe.html

# Steps to Reproduce
compile 
```
module Main

main : IO ()
main = putStrLn "Hello world"
``` 
with racket 8.3

I would suggest, add `(require racket/unsafe/ops)` to the racket prolog for 8.3 right about here - https://github.com/idris-lang/Idris2/blob/877025d5247dadd45b14aec8a9c751e93db566c5/src/Compiler/Scheme/Racket.idr#L42
but I don't see an obvious pattern for checking the require against a given racket version.

```
host % racket build/exec/hello_app/hello.rkt 
build/exec/hello_app/hello.rkt:52:20: unsafe-fx+: unbound identifier
  in: unsafe-fx+
  location...:
   build/exec/hello_app/hello.rkt:52:20
host idris2 % vi build/exec/hello_app/hello.rkt    #<- add (require racket/unsafe/ops)
host idris2 % racket build/exec/hello_app/hello.rkt
Hello world
```
# Expected Behavior
Hello world

# Observed Behavior
idris2 hello.idr -o hello         
build/exec/hello_app/hello.rkt:52:20: unsafe-fx+: unbound identifier
  in: unsafe-fx+
  location...:
   build/exec/hello_app/hello.rkt:52:20

**Comment by jfoutz** [comment]
I might be a fool. it's there in the source. I think this might only apply to the bootstrap compiler.

**Comment by jfoutz** [comment]
I think this will fix it, but it's possibly a very silly thing to do. Mainly because it'll break bootstrapping with 8.2
https://github.com/idris-lang/Idris2/pull/2281

anyway, that change works for me. I'm not in a position to easily compare with 8.2 behavior 

I realize there are no additional tests. I'd be happy to add them with some vague guidance. It's not super obvious how to, like, validate without actually running against 8.3

**Comment by jfoutz** [comment]
https://github.com/idris-lang/Idris2/pull/2286
probably a better solution, use the generated bootstrap from 8.3

**Comment by benhormann** [comment]
Apologies, the cause is my change. Try:
```
sed s/unsafe-fx// support/racket/support.rkt > "${PREFIX:-$HOME}/.idris2/idris2-0.5.1/support/racket/support.rkt"
rm -rf build/ttc libs/*/build/ttc
make IDRIS2_CG=racket
```
Or if you haven't installed the failed bootstrap:
```
sed s/unsafe-fx// support/racket/support.rkt > bootstrap-build/idris2-0.5.1/support/racket/support.rkt
sed '14i\
export IDRIS2_PREFIX="$(dirname "$DIR")"
' bootstrap/idris2-rktboot.sh > bootstrap-build/bin/idris2
rm -rf build/ttc libs/*/build/ttc
make IDRIS2_CG=racket IDRIS2_BOOT="$PWD/bootstrap-build/bin/idris2"
```

---

It shouldn't be a problem building from previous version, so let's regen/patch the bootstrap. I noticed another thing to fix from #2081, so I'll do the PR.

@bythebook The list of libraries required looks static, it'd be nice if the preamble could be moved to out to support.rkt. But this would make things even more awkward for the bootstrap process for a version or two.

**Comment by attila-lendvai** [comment]
FYI, https://github.com/idris-lang/Idris2/pull/1990 also updates the bootstrap files, fixing this at least on 8.3.

i can't easily test it on 8.2 for now.

**Comment by benhormann** [comment]
> FYI, #1990 also updates the bootstrap files, fixing this at least on 8.3.
> 
> i can't easily test it on 8.2 for now.

@jfoutz @attila-lendvai This effects all versions of Racket, the title is misleading.

**Comment by jfoutz** [comment]
Yes, my initial pr would fix racket 8.3 but break 8.2. I'm not sure if you want to support current, or current at the time of release. That's a policy decision that I don't have strong feelings about.
I briefly looked into checking versions, but didn't see anything obvious. However, if there is a constant lurking somewhere, the bootstrap could say
(if (< ver 8.3) 
    (require old-path)
    (require new-path))

or a cond or whatever.
another option would be different bootstrap sources for 8.3 and 8.2 and below, that could be detected and maintained by the makefile and config.mk

in any case, once you've got any version of idris2 up, you don't really need the bootstrap compiler anymore.

if anyone has strong feelings about the right answer, I'm happy to take a little direction and attempt another pr. I'm not sure what approach contributors would want to accept.

**Comment by jfoutz** [comment]
I was also thinking about setting up a system for building in docker containers, which would support whatever permutations are required. That's a larger undertaking, and I'm not sure the complexity is worth it. But if there's interest, I'd be willing to put something together to start the conversation.

**Comment by benhormann** [comment]
@jfoutz I repeat, there was no change in Racket! See [unsafe v8.3](https://download.racket-lang.org/releases/8.3/doc/reference/unsafe.html) vs [unsafe v8.0](https://download.racket-lang.org/releases/8.0/doc/reference/unsafe.html)
So far as I know, even v7.4 works fine.

**Comment by jfoutz** [comment]
@benhormann AHH! I see! I'd made terrible assumptions about what had changed on my machine. I'd updated racket, _and_ the nightly, I didn't dig deep bootstrap/idris2_app/idris.rkt! Sorry for the confusion. 
Thank you for your clear explanation.

Updated the title (hopefully it's clear) 
Thank you again.

**Comment by benhormann** [comment]
@CodingCellist This isn't blocked by 1990. It would be good to get #2295 merged.

**Comment by CodingCellist** [comment]
@benhormann sorry about that! I'd misunderstood the parts of the discussion in #2295 and #1990 which mentioned this.
